insertion_sort.py

```diff
@@ -0,0 +1,42 @@
+def insort(unlist):
+    """Insertion sort a list
+
+    Implementation follows after gif here
+    https://en.wikipedia.org/wiki/Insertion_sort"""
+    slist = unlist[:]
+    for i in range(1, len(slist)):
+        j = i - 1
+        while j >= 0:
+            if slist[i] < slist[j]:
+                slist[i], slist[j] = slist[j], slist[i]
+                j -= 1
+                i -= 1
+            else:
+                break
+    return slist
+
+
+if __name__ == "__main__":
+    """Test time performance for best and worst cases"""
+    import time
+    # time is better than timeit for unix times
+    # https://docs.python.org/2/library/timeit.html#timeit.default_timer
+
+    # O(n) best case
+    start = time.time()
+    for i in range(100):
+        insort(range(100))
+    stop = time.time()
+    best_time = (stop - start) / 100
+
+    # O(n**2) worst case
+    start = time.time()
+    for i in range(100):
+        insort(range(100)[::-1])
+    stop = time.time()
+    worst_time = (stop - start) / 100
+
+    print "Best case is {:.1f} times better than worst for n=100\n\n".format(
+        worst_time/best_time) +\
```

Comment:
Only use the backslash "\" line breaks if absolutely no other way exists to format your line breaks, and even then - try hard not to. See [here](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Line_length#Line_length) and [here](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Strings#Strings).